src/components/multi_select_state_utils.js

```diff
@@ -1,18 +1,36 @@
-export const findItemByIds = ids => item =>
-  ids.find(id => id === item.id) === undefined;
+const matchItemById = id => item => id === item.id;
 
-export const getSelectAllItems = (filteredItems, selectedItems, items) => {
-  const sourceItems = items.filter(
+const getItemById = (id, selectedItems) =>
+  selectedItems.find(matchItemById(id));
+
+const isFilteredEnabled = (item, filteredItems) =>
+  filteredItems.find(
+    filteredItem => item.id === filteredItem.id && !item.disabled
+  );
+
+const getSourceItems = (filteredItems, selectedItems, items) =>
+  items.filter(
     item =>
-      filteredItems.find(
-        filteredItem => item.id === filteredItem.id && !item.disabled
```

Comment:
I do see that this is something that was before our changes, but it looks like `!item.disabled` check is put inside the cycle, although it could prevent going through the cycle. Not sure if we should leave the same aproach